src/lingrex/trimming.py

```diff
@@ -1,201 +1,178 @@
 """
-Trimming functionalities in lingrex.
+Functionality to trim alignments by removing sites.
 """
 import random
+import typing
+import functools
+import itertools
 import collections
-from lingpy.sequence.sound_classes import token2class
-
 
-def revert(alms):
-    return [[row[i] for row in alms] for i in range(len(alms[0]))]        
+from lingpy.sequence.sound_classes import token2class
 
+from lingrex.util import subsequence_of
 
-def get_skeleton(alms, gap="-"):
-    return [token2class(([c for c in col if c != gap] or ["+"])[0], "cv") for col in revert(alms)]
+__all__ = ['GAP', 'Sites', 'prep_alignments']
+GAP = '-'
 
 
-def apply_trim(alms, idxs):
+class Site(list):
     """
-    Basic trimming function, based on a selection of indices that are trimmed.
+    A site in an alignment is a "column", i.e. a list of the n-th sound in the aligned words.
     """
-    return [[row[i] for i in range(len(row)) if i not in idxs] for row in alms]
+    def gap_ratio(self, gap: str = GAP) -> float:
+        return self.count(gap) / len(self)
 
+    def first_sound(self, gap=GAP):
+        for s in itertools.dropwhile(lambda c: c == gap, self):
+            return s
 
-def subsequence_of(source, target):
-    """
-    Check if source is a subsequence of target.
-    """
-    q_1, q_2 = list(target), list(source)
-    while q_1:
-        s = q_1.pop(0)
-        if q_2 and q_2[0] == s:
-            q_2.pop(0)
-        elif q_2:
-            pass
-        else:
-            break
-    if not q_2:
-        return True
-    return False
-
-
-def consecutive_gaps(seq, gap="-"):
```

Comment:
Removed in favour of using `itertools.takewhile` in the one place this was used.